st2common/st2common/services/trace.py

```diff
@@ -294,11 +297,15 @@ def get_trace_component_for_action_execution(action_execution_db):
         'ref': str(action_execution_db.action.get('ref', ''))
     }
     caused_by = {}
-    if action_execution_db.rule and action_execution_db.trigger_instance:
+    if liveaction_db and 'parent' in liveaction_db.context:
```

Comment:
I wish there was some method for obtaining a parent from a live action since I believe we do this in multiple places.